### DISCUSSION TITLE
Erratic readings

### DISCUSSION THREAD

**Discussion by edboyd40** [mKeRix/room-assistant#572] [General]
Hello,

I'm tracking two devices (iPhones) with room-assistant set up on 2 RPi Zero Ws.   I'm getting really erratic readings for Room Presence.  It flips between not_home or one of the two Pis regardless of if I'm moving or not moving.  I've tried adding/changing several of the settings I've seen in other posts, but nothing seems to be helping.   

My local.yml files are identical except for the instanceName

```
global:
  instanceName: LivingRoomPi
  integrations:
    - homeAssistant
    - bluetoothClassic
cluster:
  autoDiscovery: false
  port: 6425
  peerAddresses:
    - 192.168.86.193:6425
    - 192.168.86.194:6425
homeAssistant:
  mqttUrl: 'mqtt://192.168.86.43:1883'
  mqttOptions:
    username: mqtt
    password: mqttpwrd
bluetoothClassic:
  minRssi: -30
  hciDeviceId: 0
  interval: 5
  timeoutCycles: 3
  addresses:
    - '00:00:00:00:00:00' Phone1
    - '00:00:00:00:00:00' Phone2
entities:
  behaviors:
    bluetooth-classic-00:00:00:00:00:00-tracker:
      rollingAverage:
        window: 300
```

I'm not sure if I am missing something very basic or flubbed something up when changing around settings.   I'd like to use room presence for automations, but the readings aren't consistent enough at the moment. 

Thanks in advance.

**Comment by mKeRix** [comment]
Before we dive any deeper, two things:

1. The id you configured for your behavior is likely wrong, it should probably look more like `bluetooth-classic-00-00-00-00-00-00-tracker` - it should match what is available in the [API](https://www.room-assistant.io/guide/api.html) or logged in verbose mode
2. Have you considered using BLE with the companion app on the iPhones? That will likely result in more accurate readings and also remove some common causes of trouble from the system.

**Reply by edboyd40** [reply]
Double checked the id and it matches what is in the API.  FWIW, I had some nice solid blocks over the weekend without changing anything, so that's promising.
Added the companion app to one iPhone for comparison.   I'll test it out for a few days and see.

Thanks for your help!

**Reply by regoras** [reply]
>     it should match what is available in the [API](https://www.room-assistant.io/guide/api.html) or logged in verbose mode

Does this mean entities listed in behaviors that just use the address, ie 12_34_56_78_90 or ibeacon address is not the correct usage? I think the Entity page docs could use a bit of clarification on this point, or maybe I'm just dumb, but I saw "bluetooth-classic-xx-tracker" as an example stand-in for the device address, not necessarily as a stand in for a proper entity ID.

**Reply by edboyd40** [reply]
I’m still getting some odd readings.  Like this morning my phone dropped off while my wife’s phone did not. 

![9F19D6B4-8B94-4D98-97B3-8276E09979F6](https://user-images.githubusercontent.com/68960412/110466533-e1211b80-80a3-11eb-8c54-13c35f344e46.jpeg)

**Reply by mKeRix** [reply]
@edboyd40 Is this now for the BT Classic integration or the BLE integration (companion app)?

**Reply by edboyd40** [reply]
Bluetooth Classic.  I took one of my instances down, so I only have one running at the moment.  However for the past few days the tracker is showing the same states for both devices even if one is home and the other is away.  For instance the larger away block towards the left, I was home and my wife was not.  It’s showing both devices as away.  

I’m thinking of doing a fresh install with one instance and tracking only my device - both Bluetooth classic and BLE to see if I can get it working.  Then either add the second node or the second device.  Any advice one which to try to get working first, the node or device?  

I really appreciate your help with this. 

![D2433CE4-2137-4F92-A7D7-A532FD042E0A](https://user-images.githubusercontent.com/68960412/111144621-ab75aa00-855d-11eb-917c-fa368f95abf6.jpeg)

**Reply by mKeRix** [reply]
In your case I'd probably recommend to start with adding the second device and after that the second now. Multiple devices really shouldn't be a problem, which is why these synced states are quite confusing to me. I suppose having that on only one node would narrow down the scope for the bug hunt.

**Reply by edboyd40** [reply]
Just getting this started back up.  Tracking two devices with one node.  I'll let that run for a day and see, then add a third device.  If that all works I'll come back and add the second node in.   Thanks again for your help!
